### PR TITLE
Added CourseInstructor, CourseTopic, CoursePrice models

### LINKED ISSUE #9: Create Course sub models for Instructors, Topics, and Prices
- [x] Implement CourseInstructor
- [x] Implement CourseTopic
- [x] Implement CoursePrice

### PR DESCRIPTION
**Pull request by alizag**
#### What are the relevant tickets?
closes #9 

#### What's this PR do?
This PR introduces CourseInstructor, CourseTopic, and CoursePrice models.

#### Where should the reviewer start?
models.py

#### Any background context you want to provide?
Previously, we had JSONFields for Instructors, Topics, and Prices.  After discussion, we've concluded that it is more effective to convert these fields to be their own models instead.  This will make it easier down the line to validate the data stored in these fields.

You'll notice that I did not include `title` field for `CourseInstructor`, because the value of title is different between OCW and MITx. Here is an example that outlines the different values `title` can hold for the different platforms:
**OCW instructors object:**
```
"instructors": [
    {
        "middle_initial": "", 
        "first_name": "Paulo", 
        "last_name": "Lozano", 
        "suffix": "", 
        "title": "", 
        "mit_id": "", 
        "department": "", 
        "directory_title": "", 
        "uid": "270e2089f3504f905638185f33fbfb12"
    }, 
    {
        "middle_initial": "", 
        "first_name": "Manuel", 
        "last_name": "Martinez-Sanchez", 
        "suffix": "", 
        "title": "Martinez-Sanchez, Manuel", 
        "mit_id": "900008341", 
        "department": "Aeronautics and Astronautics", 
        "directory_title": "Professor", 
        "uid": "292dadae2775334e76d069610178bb29"
    }
], 
```
**MITx instructors object:**
```
[
    {
        "uuid": "69368f5c-5bbb-4f46-9f14-ce7db6b9dd31",
        "salutation": null,
        "given_name": "Abhijit Vinayak",
        "family_name": "Banerjee",
        "bio": "Abhijit Banerjee was educated at the University of Calcutta, Jawaharlal Nehru University, and Harvard University. He is currently the Ford Foundation International Professor of Economics at MIT. Banerjee is a past president of the Bureau for Research in the Economic Analysis of Development, a fellow of the American Academy of Arts and Sciences and the Econometric Society, and has been a Guggenheim Fellow and an Alfred P. Sloan Fellow. He is the recipient of many awards, including the inaugural Infosys Prize in 2009, and has been an honorary advisor to many organizations including the World Bank and the government of India.",
        "slug": "abhijit-vinayak-banerjee",
        "position": {
            "title": "Ford Foundation International Professor of Economics",
            "organization_name": "MIT",
            "organization_id": 27,
            "organization_override": "MIT",
            "organization_marketing_url": "https://www.edx.org/school/mitx"
        },
        "areas_of_expertise": [],
        "profile_image": {
            "medium": {
                "height": 110,
                "width": 110,
                "url": "https://prod-discovery.edx-cdn.org/media/people/profile_images/69368f5c-5bbb-4f46-9f14-ce7db6b9dd31-e82500b0b39a.medium.jpg"
            }
        },
        "works": [],
        "urls": {
            "twitter": null,
            "blog": null,
            "facebook": null
        },
        "urls_detailed": [],
        "email": null,
        "profile_image_url": "https://prod-discovery.edx-cdn.org/media/people/profile_images/69368f5c-5bbb-4f46-9f14-ce7db6b9dd31-e82500b0b39a.jpg",
        "major_works": ""
    }
],
```
I used `directory_title`(for OCW), and `position.title`(for MITx) to evaluate whether to include `title` field in `CourseInstructor` or not.  Based on the mockups provided here (https://invis.io/U5PPJZVQSAW), the only place we use `title` is in the Course description modal as a prefix to the instructor's full name.  And while it's possible to do that with the value provided by OCW instructor's object, it's not straight forward with MITx since the value doesn't seem to follow a pattern.

### CODE REVIEW

File: course_catalog/models.py [hunk 1]
```diff
@@ -18,3 +18,18 @@ class Course(models.Model):
...
     raw_json = JSONField(null=True)
+
+
+class CourseInstructor(models.Model):
+    first_name = models.CharField(max_length=128)
+    last_name = models.CharField(max_length=128)
+
+
+class CourseTopic(models.Model):
+    name = models.CharField(max_length=128)
```
Comment (mbertrand) [inline course_catalog/models.py]: make unique?

File: course_catalog/models.py [hunk 2]
```diff
@@ -18,3 +18,18 @@ class Course(models.Model):
...
+
+
+class CoursePrice(models.Model):
+    price = models.DecimalField()
+    mode = models.CharField()
+    upgrade_deadline = models.DateTimeField(null=True)
```
Comment (mbertrand) [inline course_catalog/models.py]: `courses = models.ManyToManyField(Course)`?

File: course_catalog/models.py (hunk 1 above)
Comment (mbertrand) [reply course_catalog/models.py]: `courses = models.ManyToManyField(Course)`?

File: course_catalog/models.py [hunk 3]
```diff
@@ -18,3 +18,18 @@ class Course(models.Model):
     enrollment_end = models.DateTimeField(null=True)
     image = JSONField(null=True)
     raw_json = JSONField(null=True)
+
+
+class CourseInstructor(models.Model):
+    first_name = models.CharField(max_length=128)
+    last_name = models.CharField(max_length=128)
```
Comment (mbertrand) [inline course_catalog/models.py]: Will every instructor have the `mit_id` field that could be added and used as a unique key?  Can `title` be added?

Comment (mbertrand) [reply course_catalog/models.py]: `courses = models.ManyToManyField(Course)?`

File: course_catalog/models.py [hunk 4]
```diff
@@ -18,3 +18,18 @@ class Course(models.Model):
     enrollment_end = models.DateTimeField(null=True)
     image = JSONField(null=True)
```
Comment (mbertrand) [inline course_catalog/models.py]: Forgot about this one, maybe it makes sense to make this a URLField (`src` for the image) if there's only ever one, or if there's multiple then a `CourseImage` model.

File: course_catalog/models.py (hunk 3 above)
Comment (kyfantaz) [reply course_catalog/models.py]: `title` seems difficult to ascertain apropos Ali's discussion in the PR description. What sort of values should we expect? The invision design only suggests "Prof." as a title.

Comment (mbertrand) [reply course_catalog/models.py]: ok

File: course_catalog/models.py (hunk 4 above)
Comment (mbertrand) [reply course_catalog/models.py]: Either of those options seems fine with me.

**Comment by kyfantaz** [comment]
based on all of Matt's comments, LGTM

**Pull request merged**